### PR TITLE
Incorporate terraform workflow into daemon deployment

### PR DESCRIPTION
**Pull request by xbrianh**
This runs TF workflow if terraform files are present in `daemons/{daemon_name}`

### CODE REVIEW

File: daemons/Makefile [hunk 1]
```diff
@@ -28,6 +28,16 @@ deploy-serial: $(SERIAL_AWS_DAEMONS) $(SERIAL_GCP_DAEMONS)
 deploy-parallel: $(PARALLEL_AWS_DAEMONS) $(PARALLEL_GCP_DAEMONS)
 
 $(SERIAL_AWS_DAEMONS) $(PARALLEL_AWS_DAEMONS) scheduled-ci-build:
+	@for f in $@/*.tf; do \
+		echo $$f; \
```
Comment (kislyuk) [inline daemons/Makefile]: Probably want something fancier here like `echo "Terraforming $$f"`

File: daemons/Makefile [hunk 2]
```diff
@@ -28,6 +28,16 @@ deploy-serial: $(SERIAL_AWS_DAEMONS) $(SERIAL_GCP_DAEMONS)
...
+		if [ -e $$f ]; then \
+			rm -rf $@/.terraform/*.tfstate;\
+			./build_tf_deploy_config.py $@;\
+			(cd $@ ; terraform init);\
+			(cd $@ ; TF_VAR_daemon=$@ terraform apply -auto-approve);\
+		fi;\
+		break;\
```
Comment (kislyuk) [inline daemons/Makefile]: This means only the first .tf file will be applied. Did you mean it that way, or is this a leftover debug breakpoint?

File: daemons/build_tf_deploy_config.py [hunk 3]
```diff
@@ -0,0 +1,66 @@
+#!/usr/bin/env python
```
Comment (kislyuk) [inline daemons/build_tf_deploy_config.py]: Please add a docstring here.

File: daemons/Makefile (hunk 1 above)
Comment (xbrianh) [reply daemons/Makefile]: Fixed

File: daemons/Makefile (hunk 2 above)
Comment (xbrianh) [reply daemons/Makefile]: This was intentional: once the first .tf file is detected, we know we need to perform TF workflow.

File: daemons/build_tf_deploy_config.py (hunk 3 above)
Comment (xbrianh) [reply daemons/build_tf_deploy_config.py]: Done

**Pull request merged**